src/librustc_mir/util/aggregate.rs

```diff
@@ -53,13 +53,12 @@ pub fn expand_aggregate<'tcx>(
         .map(move |(i, (op, ty))| {
             let lhs_field = if let AggregateKind::Array(_) = kind {
                 // FIXME(eddyb) `offset` should be u64.
-                let offset = i as u32;
+                let offset = i as u64;
                 assert_eq!(offset as usize, i);
                 tcx.mk_place_elem(
                     lhs,
                     ProjectionElem::ConstantIndex {
                         offset,
-                        // FIXME(eddyb) `min_length` doesn't appear to be used.
```

Comment:
the fixme is how i started working on the pr - then I realised that `min_length` is actually used [here](https://github.com/rust-lang/rust/blob/master/src/librustc_mir/borrow_check/places_conflict.rs). Hence removed the fixme because it is no longer a valid fixme xD

Reply:
gotcha :D